xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-wikimacro/xwiki-platform-rendering-wikimacro-api/src/main/java/org/xwiki/rendering/macro/wikibridge/WikiMacroParameterDescriptor.java

```diff
@@ -152,16 +134,23 @@ public String getDescription()
     @Deprecated
     public Class< ? > getType()
     {
-        return String.class;
+        Class<?> clazz;
```

Comment:
You could use ReflectionUtils#getTypeClass(Type), I had forgotten about this method.